File: ProducerConsumer/src/Solution/Producer.java
package Solution;

public class Producer extends Thread{
    private int count = 0;
    Monitor monitor;
    ProducerConsumer producerConsumer;
    Producer(Monitor monitor, ProducerConsumer producerConsumer){
        this.monitor = monitor;
        this.producerConsumer = producerConsumer;
    }

    @Override
    public void run() {
        while (!producerConsumer.getThreadFlag()){
            if(count < 3) {
                count++;
                monitor.set();
            }else {
                try {
                    Thread.sleep(1000);
                }catch (InterruptedException e) {
                }
                count = 0;
            }
        }
        System.out.println("Solution.Producer закончил работу: " + currentThread().getId());
    }
}

File: task1/src/main/kotlin/spbu/mm/parallel/programming/Producer.kt
package spbu.mm.parallel.programming

class Producer<T>(
    private val taskContainer: TaskContainer<T>,
    private val produce: () -> T,
    private val sleepTimer: Long = SLEEP_TIME,
    order: Int = 0
) : Thread(order.toString()) {
    override fun run() {
        try {
            while (!isInterrupted) {
                taskContainer.add(produce())
                sleep(sleepTimer)
            }
        } catch (ex: InterruptedException) {
            println("Producer №$name was interrupted")
        }
    }
}
